### PR TITLE
Document hetio.permute functions

### PR DESCRIPTION
**Pull request by dhimmel**
Add docstrings to hetio.permute functions.

### CODE REVIEW

**Comment by dhimmel** [comment]
@kkloste I'm going to merge this but any will address any feedback on the new documentation in a future PR if provided.

File: hetio/permute.py [hunk 1]
```diff
@@ -48,8 +82,49 @@ def permute_graph(graph, multiplier=10, seed=0, metaedge_to_excluded=dict(), log
...
+    described in https://doi.org/f3mn58. This method selects two edges and
+    attempts to swap their endpoints. If the swap would result in a valid edge,
+    the swap proceeds. Otherwise, the swap is skipped. The end result is that
+    node degree is preserved, but edges are shuffled, thereby losing their
+    original meaning.
+
+    Parameters
+    ----------
+    pair_list : list of tuples
+        List of edges to permute. Each edge is represented at a (source,
```
Comment (kkloste) [inline hetio/permute.py]: typo: `edge is represented as a`

File: hetio/permute.py [hunk 2]
```diff
@@ -48,8 +82,49 @@ def permute_graph(graph, multiplier=10, seed=0, metaedge_to_excluded=dict(), log
...
+        intermixed with (gene, compound) tuples. The only instance where order
+        of the source and target is not important is for an undirected edge
+        type where the source and target nodes are of the same type, such as
+        Gene-interacts-Gene.
+    directed : bool
+        Whether the edge should be considered directed. If False, a swap that
+        creates an a-b edge will be invalid if a b-a edge already exists.
+    multiplier : int or float
+        This is multiplied by the number of edges in pair_list to determine the
+        number of swaps to attempt.
```
Comment (kkloste) [inline hetio/permute.py]: Does this mean that the function will maker `multiplier` attempts at swapping *each* edge?
Or simply that "number of edges" * `multiplier` is the total number of attempts at swapping that will be made?

**Comment by kkloste** [comment]
@dhimmel Apologies for the delay -- I was caught up with other projects.
I've just reviewed the documentation; it looks good overall, but I made a couple small comments for typos/clarification.

Thanks for explaining all of this in detail, it's helped me understand what's going on here.

File: hetio/permute.py (hunk 2 above)
Comment (dhimmel) [reply hetio/permute.py]: > Does this mean that the function will maker multiplier attempts at swapping each edge?

Not this.

> "number of edges" * multiplier is the total number of attempts at swapping that will be made?

Yes.

File: hetio/permute.py (hunk 1 above)
Comment (dhimmel) [reply hetio/permute.py]: I'm happy to fix this, but feel free to open a PR, so you can get commit credit for spotting the typo.

Comment (dhimmel) [reply hetio/permute.py]: Refs https://github.com/hetio/hetio/pull/26. Thanks @kkloste

**Pull request merged**